### DISCUSSION TITLE
Keine Konfiguration von diesem AddOn Möglich!

### DISCUSSION THREAD

**Discussion by bm-magic** [OpenCCU/OpenCCU#1930] [🇩🇪 Fragen&Antworten]
Hallo, zu erst einen RIESEN DANK für diese Anwendung. Sie Funktioniert sehr gut, nur ich habe Probleme mit einer Anwendung.
- Ich habe auf einem NUC Über ProxMox/Home Assistant/ RaspberryMatic installiert und es funktioniert PERFEKT.
- Ich bin aber ein Schwotte, (das ist die Kreuzigung zwischen einem Schwaben und einem Schotten), und möchte (wegen dem Strom Verbrauch) die Endlösung auf einem Raspberry Pi 4 realisieren. (~1/3 Stromverbrauch).
- Nun habe ich auf meinem Pi-4 (GB) über:
a. Raspberry Pi OS Lite (64-bit)/
b. RPiOS64-IA-install.sh(pixmox)/
c. haos_generic-aarch64-8.5.qcow2
Home Assistant installiert und funktioniert. Ich habe bis jetzt File Editor und SSH Terminal installiert.
Leider bekomme ich nach der Installation und start von RaspberryMatic AddOn und dem Aufruf: "Benutzeroberfläche öffnen" etwa nach 10 sec.
die "Standartmaske von RM" (siehe Bild) -> und... ich "warte"...denn die Seite ist "eingefroren".

Bitte um Hilfe

Bohu
![Bildschirmfoto vom 2022-09-07 00-08-31](https://user-images.githubusercontent.com/78829134/188754900-4076c517-05ac-4a1d-ab06-a1818c7a1db7.png)

**Comment by bm-magic** [comment]
vorsorglich das Protokoll bei pimox:

Protokoll RM an pimox:
```
Mounting /data as /usr/local (Home Assistant Add-On): OK
Identifying host system: oci, OK
Initializing RTC Clock: onboard, OK
Running sysctl: OK
Checking for Factory Reset: not required
Checking for Backup Restore: not required
Initializing System: OK
Setup ca-certificates: OK
Starting logging: OK
Init onboard LEDs: init, OK
Starting irqbalance: OK
Starting iptables: OK
Starting network: eth0: link up, fixed, firewall, inet up, 172.30.33.0, OK
Identifying Homematic RF-Hardware: HmRF: n/a, HmIP: n/a, OK
Updating Homematic RF-Hardware: no GPIO/USB connected RF-hardware found
Starting hs485dLoader: disabled
Starting xinetd: OK
Starting eq3configd: OK
Starting lighttpd: OK
Starting ser2net: disabled
Starting ssdpd: OK
Starting ha-proxy: OK
Starting NUT services: disabled
Initializing Third-Party Addons: OK
Starting LGWFirmwareUpdate: ...OK
Setting LAN Gateway keys: OK
Starting hs485d: disabled
Starting multimacd: not required
Starting rfd: no BidCos-RF hardware found
Starting HMIPServer: .........OK
Starting ReGaHss: .OK
Starting CloudMatic: OK
Starting NeoServer: disabled
Starting Third-Party Addons: OK
find: /usr/local/etc/monit*.cfg: No such file or directory
Starting crond: OK
Setup onboard LEDs: booted, OK
Finished Boot: 3.65.8.20220831 (raspmatic_oci_arm64)
Setup onboard LEDs: shutdown, OK
Stopping crond: OK
Stopping Third-Party Addons: OK
Stopping NeoServer: OK`
Und zum Vergleich bei PxoxMox:
```

Protokoll RM im ProxMox:
```
Mounting /data as /usr/local (Home Assistant Add-On): OK
Identifying host system: oci, OK
Initializing RTC Clock: onboard, OK
Running sysctl: OK
Checking for Factory Reset: not required
Checking for Backup Restore: not required
Initializing System: OK
Setup ca-certificates: OK
Starting logging: OK
Init onboard LEDs: init, OK
Starting irqbalance: OK
Starting iptables: OK
Starting network: eth0: link up, fixed, firewall, inet up, 172.30.33.1, OK
Identifying Homematic RF-Hardware: ....HmRF: HMIP-RFUSB/eQ-3 HmIP-RFUSB@usb-0000:01:1b.0-1, HmIP: HMIP-RFUSB/eQ-3 HmIP-RFUSB@usb-0000:01:1b.0-1, OK
Updating Homematic RF-Hardware: HMIP-RFUSB: 4.4.18, not necessary, OK
Starting hs485dLoader: disabled
Starting xinetd: OK
Starting eq3configd: OK
Starting lighttpd: OK
Starting ser2net: disabled
Starting ssdpd: OK
Starting sshd: OK
Starting ha-proxy: OK
Starting NUT services: disabled
Initializing Third-Party Addons: OK
Starting LGWFirmwareUpdate: ...OK
Setting LAN Gateway keys: OK
Starting hs485d: disabled
Starting multimacd: .OK
Starting rfd: .OK
Starting HMIPServer: ........OK
Starting ReGaHss: .OK
Starting CloudMatic: OK
Starting NeoServer: OK
Starting Third-Party Addons: OK
find: /usr/local/etc/monit*.cfg: No such file or directory
Starting crond: OK
Setup onboard LEDs: booted, OK
Finished Boot: 3.65.8.20220831 (raspmatic_oci_amd64)
```

Und noch Bilder von der WM_HW:
![pimox_hw](https://user-images.githubusercontent.com/78829134/188832033-2d3c2843-2ab9-4839-aef5-308c3ad4fa9e.png)
![ProxMox_hw](https://user-images.githubusercontent.com/78829134/188832106-95eb4f2f-4f8e-42c2-b8ac-433d6ea2753f.png)

**Reply by jens-maus** [reply]
Das einzige was ich hier erkennen kann (unterschied zwischen Proxmox und Pimox ist, das das USB-Gerät unterschiedlich angebunden ist. 1x unter Proxmox mit Vendor/Product-ID und 2x unter Pimox via USB Portfreigabe. Deshalb vermute ich das es ggf. daran liegt und er unter Pimox einfach das `HmIP-RFUSB` Funkmodul deshalb beim hochfahren nicht finden kann. Wie steht es um die ausgabe von `lsusb` innerhalb des Add-ons bzw. HomeAssistant um zu sehen das das Funkmodul wirklich freigegeben wird?

Auch möchte ich noch anmerken, doch bitte einmal zu überlegen, wenn ohnehin Proxmox zum Einsatz kommt, RaspberryMatic als separate VM zu betreiben und nicht innerhalb von HomeAssistantOS als Add-on. Das sollte zusätzlich die Komplexität des ganzes wesentlich reduzieren helfen.

**Comment by Baxxy13** [comment]
Also die Geschichte finde ich spannend.
Gerade mit Pi4B-8GB (der vllt. bis auf 1,8GHz boostet) und USB3-->SSD klingt das nach interessantem Setup.

- Pimox ([pimox---proxmox-v7-for-the-raspberry-pi](https://github.com/pimox/pimox7#pimox---proxmox-v7-for-the-raspberry-pi)) scheint erstmal zu laufen. 
- [Home Assistant OS 9.0.rc1](https://github.com/home-assistant/operating-system/releases/tag/9.0.rc1) als VM (Generic aarch64) läuft auch soweit ich das sehe.

> RaspberryMatic als separate VM zu betreiben

Es geht hier um pimox, da bräuchte man erstmal eine aarch64 ova-Version, idealerweise mit Guest-Agent und allem drum und dran.

Was mir beim spielen aufgefallen ist... das pimox die HA-OS-VM weder stoppen noch rebooten noch sonst was kann. 
So ganz ausgereift ist das wohl noch nicht.

**Reply by Baxxy13** [reply]
Nee ne, das ist viel zu sehr Bastelbude. 
Scheinbar kann pimox die HB-RF-USB-2 nicht an die HA-OS-VM durchreichen.
Dazu kommt das das RM-HA-AddOn zwar laut Log fehlerfrei (nat. ohne RF-Hardware) gebootet wurde, aber Zugriff bekomme ich nicht.

**Reply by Baxxy13** [reply]
Also das USB-Passthrough habe ich noch hinbekommen.
Dafür sieht der Startvorgang des AddOns jetzt recht unschön aus und Zugriff gibt es weiterhin nicht.
```
Mounting /data as /usr/local (Home Assistant Add-On): OK
Identifying host system: oci, OK
Initializing RTC Clock: onboard, OK
Running sysctl: OK
Checking for Factory Reset: not required
Checking for Backup Restore: not required
Initializing System: OK
Setup ca-certificates: OK
Starting logging: OK
Init onboard LEDs: init, OK
Starting irqbalance: OK
Starting iptables: OK
Starting network: eth0: link up, fixed, firewall, inet up, 172.30.33.0, OK
Identifying Homematic RF-Hardware: ....HmRF: HM-MOD-RPI-PCB/HB-RF-USB-2@usb-0000:00:04.0-1, HmIP: HM-MOD-RPI-PCB/HB-RF-USB-2@usb-0000:00:04.0-1, OK
Updating Homematic RF-Hardware: HM-MOD-RPI-PCB: 2.8.6, not necessary, OK
Starting hs485dLoader: disabled
Starting xinetd: OK
Starting eq3configd: OK
Starting lighttpd: OK
Starting ser2net: disabled
Starting ssdpd: OK
Starting ha-proxy: OK
Starting NUT services: disabled
Initializing Third-Party Addons: OK
Starting LGWFirmwareUpdate: ...OK
Setting LAN Gateway keys: OK
Starting hs485d: disabled
Starting multimacd: ..........ERROR
Starting rfd: ERROR: /dev/mmd_bidcos missing, no BidCos-RF hardware found
Starting HMIPServer: ERROR: /dev/mmd_hmip missing
Starting ReGaHss: .OK
Starting CloudMatic: OK
Starting NeoServer: disabled
Starting Third-Party Addons: OK
find: /usr/local/etc/monit*.cfg: No such file or directory
Starting crond: OK
Setup onboard LEDs: booted, OK
Finished Boot: 3.65.8.20220904 (raspmatic_oci_arm64)
```

**Reply by Baxxy13** [reply]
Ich habe heute noch ein bisschen experimentiert:
pimox läuft

A:
Homeassistant-OS als VM läuft, aber ohne Guest-Agent
USB-Passtrough in die Homeassistant VM funktioniert
RM-HA-AddOn läuft nicht, bzw. Fehler beim booten (siehe voriger Post) und kein Zugriff auf die WebUI

B: 
Debian 11 VM mit Homeassistant-Supervised läuft, mit Guest-Agent
USB-Passtrough in die Debian 11 VM funktioniert
RM-HA-AddOn bootet, aber zeigt die typischen "Supervised-Probleme", RM-WebUI Zugriff funktioniert
(aber zig Restarts von HmIP-Server und 1x rfd Restart)

```
Mounting /data as /usr/local (Home Assistant Add-On): OK
Identifying host system: oci, OK
Initializing RTC Clock: onboard, OK
Running sysctl: OK
Checking for Factory Reset: not required
Checking for Backup Restore: not required
Initializing System: OK
Setup ca-certificates: OK
Starting logging: OK
Init onboard LEDs: init, OK
Starting irqbalance: OK
Starting iptables: OK
Starting network: eth0: link up, fixed, firewall, inet up, 172.30.33.0, OK
Identifying Homematic RF-Hardware: ....HmRF: HM-MOD-RPI-PCB/HB-RF-USB-2@usb-0000:00:04.0-1, HmIP: HM-MOD-RPI-PCB/HB-RF-USB-2@usb-0000:00:04.0-1, OK
Updating Homematic RF-Hardware: HM-MOD-RPI-PCB: 2.8.6, not necessary, OK
Starting hs485dLoader: disabled
Starting xinetd: OK
Starting eq3configd: OK
Starting lighttpd: OK
Starting ser2net: disabled
Starting ssdpd: OK
Starting sshd: OK
Starting ha-proxy: OK
Starting NUT services: disabled
Initializing Third-Party Addons: OK
Starting LGWFirmwareUpdate: ...OK
Setting LAN Gateway keys: OK
Starting hs485d: disabled
Starting multimacd: .OK
Starting rfd: ....................ERROR
Starting HMIPServer: .......................................................................................................................................................ERROR
Starting ReGaHss: .OK
Starting CloudMatic: OK
Starting NeoServer: disabled
Starting Third-Party Addons: OK
Starting crond: OK
Setup onboard LEDs: booted, OK
Finished Boot: 3.65.8.20220908 (raspmatic_oci_arm64)
```

**Reply by Baxxy13** [reply]
Aktueller Stand:
pimox auf Pi4B-8GB mit USB3-->SATA SSD + zusätzlicher USB-Stick für die RM-User-Daten

VM 101: debian-11.5.0-arm64 --> RaspberryMatic im Docker-Container mit HmIP-RFUSB
VM 102: debian-11.5.0-arm64 --> Homeassistant Supervised

Schnurrt soweit vor sich hin und nimmt im Normalbetrieb ca. 5W. 

Langzeiterfahrungen gibt's (klar) noch keine, und daher auch keine Empfehlung das so zu betreiben.

Persönlich würde ich gerne die Docker-RaspberryMatic durch eine RaspberryMatic-VM ersetzen.
Die gibt es aber aktuell nicht als arm64-Version und zum Feature-Request (#1937) dazu bisher auch noch kein Kommentar.

**Comment by bm-magic** [comment]
@Baxxy13, vielen Dank für Deine Veröffentlichung.
Ich versuche es die ganze Zeit, nach Deinen raren Angaben nach zubauen, aber ich kann es nicht.
-   Aus Verzweiflung habe ich die Anleitung mit DeepL ins deutsche Übersetzt, aber ich bekomme den pivccu-modul für Pi4 (8Gb) nicht installiert. Die Fehlermeldung:
- `
root@raspberryMatic:~# sudo apt install pivccu-modules-raspberrypi
Paketlisten werden gelesen… Fertig
Abhängigkeitsbaum wird aufgebaut… Fertig
Statusinformationen werden eingelesen… Fertig
Einige Pakete konnten nicht installiert werden. Das kann bedeuten, dass
Sie eine unmögliche Situation angefordert haben oder, wenn Sie die
Unstable-Distribution verwenden, dass einige erforderliche Pakete noch
nicht erstellt wurden oder Incoming noch nicht verlassen haben.
Die folgenden Informationen helfen Ihnen vielleicht, die Situation zu lösen:
Die folgenden Pakete haben unerfüllte Abhängigkeiten:
 pivccu-modules-raspberrypi : Hängt ab von: raspberrypi-kernel-headers ist aber nicht installierbar oder
                                             linux-headers-raspi ist aber nicht installierbar
E: Probleme können nicht korrigiert werden, Sie haben zurückgehaltene defekte Pakete.
root@raspberryMatic:~#`
Wie hast Du es BITTE installiert?
Kannst Du Bitte Deine Befehle in meine *.txt Dateien korigieren? Vielen Dank im Voraus.

[Debi_auf_pimoxs.txt](https://github.com/jens-maus/RaspberryMatic/files/9733559/Debi_auf_pimoxs.txt)
[Debian_Docker_OCI_RaspberryMatic_s.txt](https://github.com/jens-maus/RaspberryMatic/files/9733561/Debian_Docker_OCI_RaspberryMatic_s.txt)

**Reply by Baxxy13** [reply]
Sorry wenn ich das so hart sagen muss, aber wenn man solch ein Konstrukt nicht selbstständig fehlerfrei aufsetzen kann dann sollte man die Finger von lassen.
Ich würde das aktuell so niemals für den produktiven Einsatz empfehlen und kenne auch keinen der das so nutzt.
Selbst wenn man es zum laufen bekommt steht man alleine da wenn es zu Problemen kommt.
Tut mir leid, aber ich werde keine Step-by-Step Anleitung erstellen.

Das...
`sudo apt install pivccu-modules-raspberrypi`
...brauchst du auch gar nicht. Zum einen ist eine VM nicht mehr wirklich ein Raspi und zum anderen hast du kein GPIO-Funkmodul (welches sich vermutlich eh nicht durchreichen lässt), sondern eine USB-Funklösung (HmIP-RFUSB).

Wenn Jens vielleicht irgendwann eine .ova für diese Architektur rausbringt wird einiges einfacher, dann kann man nochmal drüber reden.

**Comment by bm-magic** [comment]
@Baxxy13, vielen DANK für die Antwort. Es wäre ein schönes Projekt. Ich habe dafür zwei Monate meines Leben geopfert...:-(
Aber, ich muss zugeben, habe auch sehr viel gelernt...es bleibt zu hoffen, dass ich es noch erlebe, wenn Jens eine .ova arch64 herausbringt. Du kannst ihm doch dabei helfen. Oder?
Die RaspberryMatic AdOns Home Assistant auf Pi4 funktioniert bei mir zuverlässig. Hut ab und Dank dafür!
- Nur...die Schikane mit pimox wäre natürlich viel schöner zu verwalten. Zu dem, wenn das System konfiguriert ist, tut sich der Pi4 (8Gb)
sehr "langweilen" und man könnte es noch für andere Aufgaben "missbrauchen".
- Ich denke z.B. an "Pihole" und/oder eigenes DNS Server...
Na ja, man darf doch träumen...und..."die Hoffnung stirb zu Letzt".
- Aber, Du hasst absolut Recht, "lieber ein Ende mit Schrecken, als Schrecken ohne Ende."
- Darf ich noch eine Frage stellen? In den AdOns sehe ich zwei RM Versionen. Wo/was ist der Unterschied?
- Sehr Nette Grüße
Bohu

**Reply by Baxxy13** [reply]
Ja, mir gefällt die Idee auch.
Zumal gerade der Pi4B-8GB genug Dampf (durch den Boost auf bis zu 1,8GHz) und RAM für mehrere VM's zur Verfügung hat. Gebootet von SSD hat man auch wenig Sorgen bezüglich SD-Karten Verschleiß.

Pimox läuft sehr gut soweit ich das bisher getestet habe. 
Homeassistant-OS/Supervised rennt auch als VM.
Nur RaspberryMatic als Docker liegt mir nicht, funktionieren tut es aber.

"Helfen" kann ich Jens nicht wirklich. Da fehlt mir die Expertise. Aber testen tue ich alles was angeboten wird.

Ich habe mir inzwischen so einen China-PC mit Celeron J4125 zugelegt und darauf Proxmox samt RaspberryMatic-VM und Homeassistant-VM installiert. Nimmt auch nur ~6W und liegt damit nur ~1W über meinem Pi4B. Dadurch das es X86-Hardware ist rennt halt alles ohne Komplikationen.
Da ich aber "Pi4B - Fan" bin bleibe ich natürlich am Ball.

Um die Frage noch zu beantworten:
Das Eine ist die "offizielle RaspberryMatic" die etwa monatlich released wird, das Andere sind die "Nightly-Snapshots" die manchmal täglich released werden und Fixes / Optimierungen / Erweiterungen enthalten aber eher für reine Testumgebungen gedacht sind.

**Comment by bm-magic** [comment]
@Baxxy13, DANKE für die superschnelle Antwort. (Wie gewohnt ohne Details :-)  )
Natürlich hätte mich interessiert, wie man die "rechte" Option  also die "Nightly-Snapshots" installiert.
Aber, eigentlich kann ich froh sein, das es mir gelungen ist, die "offizielle" Version zu installieren, denn
ich habe als Newbie schon genug andere Probleme.
Nun bin ich gerade dabei, die SSD mit pimox mit Clonezilla zu "konservieren" um sie u.U. noch verwenden zu können,
und die "normale" Home Assistant Pi Installation zu wiederholen...habe inzwischen genug Erfahrung gesammelt. :-)
Nette Grüße
Bohu
PS Darf ich nach der Quelle für den China-PC fragen? Ich habe einen funktionierenden NUC, aber 15 bis 20 W/h sind mir
für den Dauereinsatz "echt" zu viel.

**Reply by Baxxy13** [reply]
Die beiden AddOn-Versionen werden doch genau gleich installiert.
AddOn-Store öffnen, AddOn anklicken und installieren wählen. Mann kann auch beide gleichzeitig installiert haben.

Es ist die CHUWI Herobox (Mini PC Intel Celeron J4125), ich wollte was Lüfterloses. War zum Schulstart im Angebot für etwa 160€. Findet man im großen Online Versandhaus mit "A".

**Comment by bm-magic** [comment]
@Baxxy13, danke für die info und die Details...wird sich etwa in 5 Jahren amortisieren.
Dafür tue ich es lieber noch einmal mit der "normalen" Home Assistant für Pi4 versuchen...
falls ich damit wieder (selbst gemachte) Probleme bekomme, werde ich provisorisch den NUX Proxmox verwenden
und hoffen, das bis dahin die arch64 "serienreif" wird.
...Wobei, die Raspberry Matic lässt sich achon nun starten und auch die Geräte lassen sich aus Backup einfügen,
leider ist der USB Stick nicht im Posteingang. :-(
